content/hardware/07.opta/opta-family/opta/tutorials/18.conveyor-sorting-app-note/content.md

```diff
@@ -0,0 +1,621 @@
+  ArduinoCloud.addProperty(cloudRunningSystem, READ, ON_CHANGE, NULL);
+  ArduinoCloud.addProperty(cloudRedSorted, READ, ON_CHANGE, NULL);
+  ArduinoCloud.addProperty(cloudGreenSorted, READ, ON_CHANGE, NULL);
+  ArduinoCloud.addProperty(cloudBlueSorted, READ, ON_CHANGE, NULL);
+}
+
+/*
+  Callback function triggered when the cloud start button value changes.
+  This function updates the local PLC input corresponding to the start button.
+*/
+void onStartButtonChange()  {
+  // Add your code here to act upon CloudButton change
+  PLCIn.in_cloudStartButton = cloudStartButton;
+}
+
+/*
+  Callback function triggered when the cloud stop button value changes.
+  This function updates the local PLC input corresponding to the stop button.
+*/
+void onStopButtonChange()  {
+  // Add your code here to act upon CloudButton change
+  PLCIn.in_cloudStopButton = cloudStopButton;
+}
+```
+
+**Explanation:**
+
+1. **`initProperties` Function**:
+   - `ArduinoCloud.update();`:This function links the cloud variables with their respective properties in the Arduino Cloud. Each property is set with specific attributes:
+     - **`READWRITE`**: The property can be read from and written to by the cloud.
+     - **`READ`**: The property can only be read by the cloud.
+     - **`ON_CHANGE`**: The associated function (callback) is called whenever the value of the property changes.
+   - `onStartButtonChange` and `onStopButtonChange` are callback functions that execute whenever the `cloudStartButton` or `cloudStopButton` values change.
+
+2. **Callback Functions**:
+   - **`onStartButtonChange()`**: Updates the local start button state (`PLCIn.in_cloudStartButton`) whenever the `cloudStartButton` value changes in the cloud.
+   - **`onStopButtonChange()`**: Updates the local stop button state (`PLCIn.in_cloudStopButton`) whenever the `cloudStopButton` value changes in the cloud.
+
+***You can download the complete code from [here](assets/ConveyorSortingSystem.zip).***
+
+### Arduino Cloud Dashboard
+
+The Arduino Cloud provides a powerful platform for monitoring and controlling the conveyor and sorting system in real time. With this setup, you can remotely oversee the entire process and interact with key components of the system as they happen. The system tracks the total number of items sorted by color, red, green, and blue, allowing you to monitor how many of each have been processed.
+
+In addition to tracking the sorted items, the Arduino Cloud also allows you to control the start and stop of the system. You can initiate or stop the system directly from the cloud interface, giving you full control over the operation from anywhere with an internet connection, all in real time.
+
+This real-time integration ensures that you can both observe the performance of your sorting system and take direct action when needed, making it a versatile and effective tool for managing your operations instantly and efficiently.
+
+![Arduino Cloud Dashboard](assets/conveyor-dashboard.gif)
```

Comment:
Let's give you access to an enterprise Arduino Cloud plan so you can add more widgets and customize the dashboard.